prjxray/util.py

```diff
@@ -172,43 +172,34 @@ def addr2btype(base_addr):
     return block_type_i2s[block_type_i]
 
 
-def gen_tile_bits(db_root, tilej, strict=False, verbose=False):
+def gen_tile_bits(tile_segbits, tile, strict=False, verbose=False):
     '''
-    For given tile yield
+    For given tile and corresponding db_file structure yield
     (absolute address, absolute FDRI bit offset, tag)
 
-    For each address space
-    Find applicable files
-    For each tag bit in those files, calculate absolute address and bit offsets
+    db_file:
+    {'<block_type_1>': [<bit lines>], '<block_type_2>': [<bit lines>], ...}
 
-    Sample file names:
-    segbits_clbll_l.db
-    segbits_int_l.db
-    segbits_bram_l.block_ram.db
+    For each tag bit in the corresponding block_type entry, calculate absolute address and bit offsets
     '''
-    for block_type, blockj in tilej["bits"].items():
-        baseaddr = int(blockj["baseaddr"], 0)
-        bitbase = 32 * blockj["offset"]
-
-        if block_type == "CLB_IO_CLK":
-            fn = "%s/segbits_%s.db" % (db_root, tilej["type"].lower())
-        else:
-            fn = "%s/segbits_%s.db.%s" % (
-                db_root, tilej["type"].lower(), block_type.lower())
-        # tilegrid runs a lot earlier than fuzzers
-        # may not have been created yet
-        verbose and print("Check %s: %s" % (fn, os.path.exists(fn)))
-        if strict:
-            assert os.path.exists(fn)
-        elif not os.path.exists(fn):
-            continue
-
-        for line, (tag, bits, mode) in parse_db_lines(fn):
-            assert mode is None
-            for bitstr in bits:
+
+    for block_type in tile_segbits:
+        assert block_type.value in tile[
+            "bits"], "block type %s is not present in current tile" % block_type.value
```

Comment:
Get rid of .value for block_type, not required.

Reply:
Actually I think it is necessary, the block type in tile_segbits comes from an `enum`. If i do not get its value, this error will be produced.
 
```
Traceback (most recent call last):
  File "utils/checkdb.py", line 159, in <module>
    main()
  File "utils/checkdb.py", line 155, in main
    run(args.db_root, strict=args.strict, verbose=args.verbose)
  File "utils/checkdb.py", line 141, in run
    check_tile_overlap(db, db_root, strict=strict, verbose=verbose)
  File "utils/checkdb.py", line 96, in check_tile_overlap
    verbose=verbose)
  File "utils/checkdb.py", line 34, in make_tile_mask
    tile_segbits, tilej, strict=strict, verbose=verbose):
  File "/home/alessandro/projects/symbiflow/tmp/prjxray/env/lib/python3.6/site-packages/prjxray/util.py", line 188, in gen_tile_bits
    "bits"], "block type %s is not present in current tile" % block_type
AssertionError: block type BlockType.CLB_IO_CLK is not present in current tile
```

Reply:
Ya, that's because you are using the wrong data structure here.  Rather than using the tile, get it from Grid.gridinfo_at_tilename()

Reply:
Done